apps/src/templates/rubrics/LearningGoal.jsx

```diff
@@ -38,6 +58,87 @@ export default function LearningGoal({
     setIsOpen(!isOpen);
   };
 
+  const handleFeedbackChange = event => {
+    if (studentLevelInfo.user_id && learningGoal.id) {
+      if (autosaveTimer.current) {
+        clearTimeout(autosaveTimer.current);
+      }
+      teacherFeedback.current = event.target.value;
+      setDisplayFeedback(teacherFeedback.current);
+      autosaveTimer.current = setTimeout(() => {
+        autosave();
+      }, saveAfter);
+    }
+  };
+
+  const autosave = () => {
+    console.log('autosaving');
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoal.id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    fetch('/get_token').then(response => {
+      fetch(`${base_endpoint}/${learningGoalEval.id}`, {
+        method: 'PUT',
```

Comment:
I wonder if it's worth adding a method to `HttpClient` to `put`? It would be nice to not have to call `/get_token` multiple times

Reply:
agreed